packages/autocomplete-core/src/onInput.ts

```diff
@@ -69,7 +74,13 @@ export function onInput<TItem extends BaseItem>({
     // promises to keep late resolving promises from "cancelling" the state
     // updates performed in this code path.
     // We chain with a void promise to respect `onInput`'s expected return type.
-    return runConcurrentSafePromise(collections).then(() => Promise.resolve());
+    const request = cancelable(
+      runConcurrentSafePromise(collections).then(() => Promise.resolve())
+    );
+
+    store.pendingRequests.add(request);
+
+    return request;
```

Comment:
Below we use this syntax. Do we want to do the same here?

```suggestion
    return store.pendingRequests.add(request);
```